plugin/server_audit/server_audit.cc

```diff
@@ -340,6 +353,13 @@ struct connection_info
   int proxy_host_length;
 };
 
+std::vector<std::string> log_queue(1);
```

Comment:
This is not a good idea.  Instead of adding every sting to a vector, I would rather see one big buffer where strings are concatenated e.g.

  // We are using a pointer to string to guarantee that we can decrease the size
  // of the buffer
  using data_t = std::unique_ptr<std::string>;
  // memory buffer storing the log messages
  data_t data;

and then when you reserve it

  data_t new_buffer(new std::string());
  new_buffer->reserve(buffer_size);
  data.swap(new_buffer);